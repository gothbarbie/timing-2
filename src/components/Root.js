import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'

import defaultStore from '../store/configureStore'

const Root = ({ store, children }) => (
  <Provider store={store ? store : defaultStore}>{children}</Provider>
)

Root.propTypes = {
  store: PropTypes.shape({}),
  children: PropTypes.node,
}

export default Root
