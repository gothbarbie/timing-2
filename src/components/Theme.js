import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
