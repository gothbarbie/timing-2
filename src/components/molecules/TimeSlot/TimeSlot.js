import React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'

import { openModal } from 'generic/molecules/Modal/modalActions'

import Button from 'generic/atoms/Button'
import Icon from 'generic/atoms/Icon/Icon'
import IconButton from 'generic/atoms/IconButton'

const TimeSlotStyle = styled.div`
  display: flex;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.lightGray};
  background-color: ${({ blue, theme }) => blue && theme.colors.primaryLight};

  height: 5rem;
  padding: 0 1rem;
`

const Title = styled.div`
  font-size: 1.2rem;
  font-weight: 400;
`

const TimeSlot = ({ data, openModal }) => {
  if (data) {
    return (
      <TimeSlotStyle blue>
        <Title>{data.title}</Title>
        <IconButton
          color="gray"
          icon={['fas', 'ellipsis-h']}
          onClick={() => openModal({ content: 'editEvent' })}
        />
      </TimeSlotStyle>
    )
  }
  return (
    <TimeSlotStyle>
      <Button
        fullWidth
        hasIcon
        icon={['fas', 'plus-circle']}
        onClick={() => openModal({ content: 'addEvent' })}
      >
        <Icon icon={['fas', 'plus-circle']} size="lg" />
      </Button>
    </TimeSlotStyle>
  )
}

const mapDispatchToProps = {
  openModal,
}

export default connect(
  null,
  mapDispatchToProps
)(TimeSlot)
