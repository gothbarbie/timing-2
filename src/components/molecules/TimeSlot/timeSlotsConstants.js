export const SET_TIMESLOTS = 'SET_TIMESLOTS'
