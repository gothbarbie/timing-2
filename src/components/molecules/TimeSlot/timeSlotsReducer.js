import { SET_TIMESLOTS } from './timeSlotsConstants'

const initialState = {}

const timeSlotsReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_TIMESLOTS:
      return action.payload

    default:
      return state
  }
}

export default timeSlotsReducer
