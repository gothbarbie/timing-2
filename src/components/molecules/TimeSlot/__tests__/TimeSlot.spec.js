import React from 'react'
import { render } from 'enzyme'

import Root from '../../../Root'
import Theme from '../../../Theme'
import TimeSlot from '../TimeSlot'

describe('components/molecules/TimeSlot', () => {
  it('renders with data', () => {
    const props = {
      data: {
        title: 'TimeSlot Test',
      },
      openModal: jest.fn(),
    }
    const component = render(
      <Root>
        <Theme>
          <TimeSlot {...props} />
        </Theme>
      </Root>
    )

    expect(component).toMatchSnapshot()
  })

  it('renders without data', () => {
    const props = {
      openModal: jest.fn(),
    }
    const component = render(
      <Root>
        <Theme>
          <TimeSlot {...props} />
        </Theme>
      </Root>
    )

    expect(component).toMatchSnapshot()
  })
})
