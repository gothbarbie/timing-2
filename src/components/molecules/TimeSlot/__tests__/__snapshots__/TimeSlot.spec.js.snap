// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/TimeSlot renders with data 1`] = `
<div
  class="TimeSlotStyle-xpn2pf-0 iwJCDl"
>
  <div
    class="Title-xpn2pf-1 iEbIfR"
  >
    TimeSlot Test
  </div>
  <button
    class="ButtonStyle-sc-1vcx5rr-0 efKZGt"
    color="gray"
  />
</div>
`;

exports[`components/molecules/TimeSlot renders without data 1`] = `
<div
  class="TimeSlotStyle-xpn2pf-0 kqNPtK"
>
  <button
    class="WithIcon-sc-12a0ki3-1 gTgDPM"
  />
</div>
`;
